ci/build_common.sh

```diff
@@ -1,19 +1,52 @@
 #!/bin/bash
 
-set -euo pipefail
+set -eo pipefail
 
 # Ensure the script is being executed in its containing directory
 cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )";
 
+# Script defaults
+CUDA_COMPILER=nvcc
+
 # Check if the correct number of arguments has been provided
-if [ "$#" -ne 3 ]; then
-    echo "Usage: $0 <HOST_COMPILER> <CXX_STANDARD> <GPU_ARCHS>"
+function usage {
+    echo "Usage: $0 [OPTIONS] <HOST_COMPILER> <CXX_STANDARD> <GPU_ARCHS>"
     echo "The PARALLEL_LEVEL environment variable controls the amount of build parallelism. Default is the number of cores."
     echo "Example: PARALLEL_LEVEL=8 $0 g++-8 14 \"70\" "
     echo "Example: $0 clang++-8 17 \"70;75;80-virtual\" "
+    echo "Possible options: "
+    echo "  -nvcc: path/to/nvcc"
```

Comment:
What do you think about making this option more general and calling it `--cuda_compiler` to enable us to later set up clang-cuda builds?

Reply:
> What do you think about making this option more general and calling it `--cuda_compiler` to enable us to later set up clang-cuda builds?

I didn't do that only because there was some existing NVCC logic that comes later.

Reply:
Yeah, I guess we'll just cross that bridge when we get there.